### PR TITLE
refactor: Reorganize the project structure

### PR DESCRIPTION
**Pull request by vadorovsky**
* Rename programs - these names are going to be used on crates.io:
  * `merkle_tree_program` -> `light-merkle-tree-program`
  * `verifier_program_zero` -> `light-psp2in2out`
  * `verifier_program_one` -> `light-psp10in2out`
  * `verifier_program_two` -> `light-psp4in4out`
  * `verifier_program_storage` -> `light-psp2in2out-storage`
* Merge all Rust crates into one workspace.
* Rename `system-programs` to `programs` - unfortunately, when moving `Cargo.toml` and `Anchor.toml` to the top, Anchor was struggling with finding program crates anywhere outside `programs/*`.
* Stricter rustfmt rules.

### CODE REVIEW

File: psp-examples/swap/circuits/swaps/swaps.circom [hunk 1]
```diff
@@ -0,0 +1,335 @@
+/**
```
Comment (vadorovsky) [inline psp-examples/swap/circuits/swaps/swaps.circom:1]: This file is unrelated to the goal of this PR, but I think it's a good change to make? Seems like this circuit was never commited, even though it's clearly produced by macro-circum whenever the swap PSP is built.

**Review by ananas-block — CHANGES_REQUESTED** [review]
This naming is also not consistent anymore but could also be adapted in a future pr:
https://github.com/Lightprotocol/light-protocol/blob/main/zk.js/src/utils.ts#L12

File: programs/package.json
[the remark this answers is not in the document]
Comment (vadorovsky) [reply programs/package.json:1]: Yes - this `programs` package needs to exist, so we can depend on it in other packages, like cli:

https://github.com/Lightprotocol/light-protocol/pull/318/files#diff-088b5e0ac52dd36816ebc6b121d5423c763c3c18c098b91757c014937c7d632dR47

This way we always make sure that programs are built before.

File: circuit-lib/circuit-lib.circom/scripts/checkBuildCircuit.sh [hunk 2]
```diff
@@ -38,11 +38,11 @@ check_file_modified() {
...
 }
 
-check_file_modified ../../system-programs/programs/verifier_program_zero/src/verifying_key.rs
-check_file_modified ../../system-programs/programs/verifier_program_storage/src/verifying_key.rs
-check_file_modified ../../system-programs/programs/verifier_program_two/src/verifying_key.rs
-check_file_modified ../../system-programs/programs/verifier_program_one/src/verifying_key.rs
+check_file_modified ../../programs/verifier_program_zero/src/verifying_key.rs
+check_file_modified ../../programs/verifier_program_storage/src/verifying_key.rs
+check_file_modified ../../programs/verifier_program_two/src/verifying_key.rs
+check_file_modified ../../programs/verifier_program_one/src/verifying_key.rs
```
[the remark this answers is not in the document]
Comment (vadorovsky) [reply circuit-lib/circuit-lib.circom/scripts/checkBuildCircuit.sh]: Done,

File: programs/package.json
[the remark this answers is not in the document]
Comment (ananas-block) [reply programs/package.json:1]: Ah I see

**Pull request merged**